packages/amplify-graphql-auth-transformer/src/__tests__/owner-auth.test.ts

```diff
@@ -937,4 +937,171 @@ describe('owner based @auth', () => {
+      );
+    });
+
+    test('handles sortKeyFields with explicit single claim and explicit identity claim', () => {
+      const schema = `
+        type Test @model @auth(rules: [{ allow: owner, identityClaim: "username" }]) {
+          id: ID! @primaryKey(sortKeyFields: "owner")
+          owner: String!
+        }`;
+
+      const transformer = new GraphQLTransform({
+        authConfig,
+        transformers: [
+          new ModelTransformer(),
+          new PrimaryKeyTransformer(),
+          new AuthTransformer(),
+        ],
+        featureFlags: ({
+          getBoolean: (featureName: string, defaultValue: boolean) => {
+            if (featureName === 'useSubUsernameForDefaultIdentityClaim') {
+              return true;
+            }
+
+            return defaultValue;
+          },
+        } as FeatureFlagProvider),
+      });
+
+      const out = transformer.transform(schema);
+      expect(out).toBeDefined();
+    });
+
+    test('handles sortKeyFields with implicit single claim and use username for default', () => {
+      const schema = `
+        type Test @model @auth(rules: [{ allow: owner }]) {
+          id: ID! @primaryKey(sortKeyFields: "owner")
+          owner: String!
+        }`;
+
+      const transformer = new GraphQLTransform({
+        authConfig,
+        transformers: [
+          new ModelTransformer(),
+          new PrimaryKeyTransformer(),
+          new AuthTransformer(),
+        ],
+        featureFlags: ({
+          getBoolean: (featureName: string, defaultValue: boolean) => {
+            if (featureName === 'useSubUsernameForDefaultIdentityClaim') {
+              return false;
+            }
+
+            return defaultValue;
+          },
+        } as FeatureFlagProvider),
+      });
+
+      const out = transformer.transform(schema);
+      expect(out).toBeDefined();
+    });
+
+    test('handles sortKeyFields with implicit single claim, custom owner field, and use sub::username for default', () => {
+      const schema = `
+        type Test @model @auth(rules: [{ allow: owner, ownerField: "myOwnerField" }]) {
+          id: ID! @primaryKey(sortKeyFields: "myOwnerField")
+          myOwnerField: String!
+        }`;
+
+      const transformer = new GraphQLTransform({
+        authConfig,
+        transformers: [
+          new ModelTransformer(),
+          new PrimaryKeyTransformer(),
+          new AuthTransformer(),
+        ],
+        featureFlags: ({
+          getBoolean: (featureName: string, defaultValue: boolean) => {
+            if (featureName === 'useSubUsernameForDefaultIdentityClaim') {
+              return true;
+            }
+
+            return defaultValue;
+          },
+        } as FeatureFlagProvider),
+      });
+
+      expect(() => {
+        transformer.transform(schema);
+      }).toThrow(
+        "The primary key's sort key type 'myOwnerField' cannot be used as an owner @auth field too. Please use another field for the sort key.",
+      );
+    });
+  });
```

Comment:
Originally I had put these tests in the index transformer because the validation seemed to fit better in the sort key validations that run. There was a problem with circular dependencies though during the build, so I moved these tests to the auth transformer to prevent the circular dependency (I believe it was index transformer -> auth -> relational -> index). I can create a follow up task to organize these tests within their module, or move them to the velocity tests.